File: src/commands/utility/checkIP.js
const Discord = require("discord.js");
const util = require("../../util");
const { ping } = require("minecraft-protocol");

module.exports = {
  args: true,
  name: "checkip",
  aliases: ["checkserverip"],
  description: "Checks if an Aternos-IP is free to use.",
  
  async execute(message, args, client) {
    function removeColorsFromString(text) {
        // Removing minecraft colors from strings, because console can`t read it and it will look crazy.
        return text.replace(/§./g, "");
    }
    let ip = args[0].match(/^(\w+)(?:\.aternos\.me)?$/i);
    if (!ip) {
        await message.reply(`\`${args}\` is not a valid Aternos server IP or name.`);
        return;
    }
    ip = ip[1];

    const embed = new Discord.MessageEmbed()
    const test = await ping({ host: `${ip}.aternos.me` });
    if(removeColorsFromString(test.description.match(/(Server not found)/gi))){
        embed.setDescription(`**${ip}.aternos.me** is free to use!\nTo use it as your server address, head to **[the options of your server](https://aternos.org/options)**`)
        embed.setColor(util.color.green)
    } else {
        embed.setDescription(`It seems a server with the IP **${ip}.aternos.me** already exists!`)
        embed.setColor(util.color.red)
    }

    await message.reply({ embeds: [embed] });
  },
};

File: py/cogs/checkip.py
from discord.ext import commands
from discord import Embed
from utils.data.utils import utils
import re
from mcstatus import JavaServer
class CheckIP(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
    
    @commands.command(aliases=["checkserverip", "check"], description = "Checks if an Aternos-IP is free to use.")
    async def checkip(self, ctx, ip):
        if ip is None:
            return await ctx.reply("Please provide a Aternos server ip!\nExample: example.aternos.me")
        if not re.match(r"^(\w+)(?:\.aternos\.me)?$", ip):
            return await ctx.reply(f"`{ip}`is not a valid Aternos server IP or name.")
        if ip.count(".") > 2:
            return await ctx.reply("Please provide a valid Aternos server ip!\nExample: example.aternos.me")
        nip = ip.split(".")[0]
        if len(nip) > 20:
            return await ctx.reply("Aternos IPs can only be 20 characters long, please try a shorter one. Yours is " + str(len(nip)) + " characters long.")
        if len(nip) < 4:
            return await ctx.reply("Aternos IPs must be at least 4 characters long, please try a longer one. Yours is " + str(len(nip)) + " characters long.")
        async with ctx.typing():
            embed = Embed()
            server = await JavaServer.async_lookup(ip)
            stat = await server.async_status()
            if stat.version.name == "⚠ Error":
                embed.description=f"**{nip}.aternos.me** is free to use!\nTo use it as your server address, head to **[the options of your server](https://aternos.org/options)**"
                embed.colour = utils.colors.green
            else:
                embed.description=f"**{nip}.aternos.me** is already taken!"
                embed.colour = utils.colors.red
            await ctx.reply (embed=embed)


def setup(bot):
    bot.add_cog(CheckIP(bot))